src/gwt/src/org/rstudio/core/client/hyperlink/RunHyperlink.java

```diff
@@ -0,0 +1,143 @@
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.core.client.hyperlink;
+
+import com.google.gwt.user.client.ui.HTML;
+import com.google.gwt.user.client.ui.Label;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import com.google.gwt.user.client.ui.Widget;
+
+import org.rstudio.core.client.StringUtil;
+import org.rstudio.core.client.regex.Match;
+import org.rstudio.core.client.regex.Pattern;
+import org.rstudio.studio.client.RStudioGinjector;
+import org.rstudio.studio.client.application.events.EventBus;
+import org.rstudio.studio.client.common.codetools.RCompletionType;
+import org.rstudio.studio.client.server.Server;
+import org.rstudio.studio.client.server.ServerError;
+import org.rstudio.studio.client.server.ServerRequestCallback;
+import org.rstudio.studio.client.workbench.views.console.events.SendToConsoleEvent;
+import org.rstudio.studio.client.workbench.views.help.model.HelpInfo;
+
+public class RunHyperlink extends Hyperlink
+{
+    public RunHyperlink(String url, String params, String text, String clazz)
+    {
+        super(url, params, text, clazz);
+        code_ = url.replaceFirst("^(ide|rstudio):run:", "");
+        Match match = HYPERLINK_PATTERN.match(code_, 0);
+        if (match == null)
+        {
+            package_ = null;
+            allowed_ = false;
+        }
+        else 
+        {
+            package_ = match.getGroup(1);
+            fun_ = match.getGroup(2);
+            allowed_ = true;
+        }
+        server_ = RStudioGinjector.INSTANCE.getServer();
+    }
+
+    @Override
+    public String getAnchorClass() 
+    {
+        return (package_ == null) ? styles_.xtermUnsupportedHyperlink() : styles_.xtermCommand();
+    }
+
+    @Override 
+    public void onClick(){
+        if (package_ != null)
+        {
+            server_.getPackageHyperlinkRisk(package_, new ServerRequestCallback<String>(){
+
+                @Override
+                public void onResponseReceived(String response) {
+                    if (StringUtil.equals(response, "loaded") || StringUtil.equals(response, "allowed"))
+                    {
+                        // package is loaded or in the allow list (e.g. testthat)
+                        events_.fireEvent(new SendToConsoleEvent(code_, true));
+                    }
+                    else if (StringUtil.equals(response, "installed")) 
+                    {
+                        // package is installed but not loaded
+                        events_.fireEvent(new SendToConsoleEvent(code_, false));
+                    }
+                    else
+                    {
+                        // package is not installed, make the code a comment
+                        // maybe this should offer to install it ?
+                        events_.fireEvent(new SendToConsoleEvent("# " + code_, false));
```

Comment:
I'm not quite sure I see the value of copying the code to the console as a comment -- it feels a little obscure for the user (would rather just have it copied to the console and if they try to run it they get the more informative error about `no package called xxx`

Reply:
oh yeah that makes much more sense.

Reply:
Changed the rpc so that it returns a boolean: is this package safe for hyperlink ? i.e. is it loaded or part of the internal allow list: devtools, usethis, testthat, rlang